File: gen_tools/gen_image.py
import urllib.request, urllib.error, json, base64, sys, os, traceback

SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
LOG = os.path.join(SCRIPT_DIR, "gen_image.log")

def log(msg):
    print(msg)
    with open(LOG, "a", encoding="utf-8") as f:
        f.write(msg + "\n")

open(LOG, "w").close()
log("=== gen_image.py start ===")

KEY    = os.environ.get("GAPI_KEY") or "AIzaSyC7umcmAdwscOCeBw8xeytqyABW_FJ9YiQ"
MODEL  = os.environ.get("GAPI_MODEL", "imagen-4.0-fast-generate-001")
PROMPT = os.environ.get("GAPI_PROMPT", "a golden trophy cup for horse racing arcade game, flat design icon, gold color, clean vector style")
OUT    = os.environ.get("GAPI_OUT", os.path.join(SCRIPT_DIR, "test_trophy.png"))

log(f"MODEL : {MODEL}")
log(f"PROMPT: {PROMPT}")
log(f"OUT   : {OUT}")

url  = f"https://generativelanguage.googleapis.com/v1beta/models/{MODEL}:predict?key={KEY}"
body = json.dumps({
    "instances": [{"prompt": PROMPT}],
    "parameters": {"sampleCount": 1}
}).encode()

req = urllib.request.Request(url, data=body, headers={"Content-Type": "application/json"})
try:
    log("Sending request...")
    with urllib.request.urlopen(req, timeout=30) as resp:
        raw = resp.read()
    log(f"Response length: {len(raw)} bytes")
    data = json.loads(raw)
    if "predictions" in data:
        img = base64.b64decode(data["predictions"][0]["bytesBase64Encoded"])
        with open(OUT, "wb") as f:
            f.write(img)
        log(f"OK - saved {OUT} ({len(img)} bytes)")
    else:
        log("API ERROR: " + json.dumps(data, indent=2))
except urllib.error.HTTPError as e:
    log(f"HTTPError {e.code}: {e.read().decode()}")
except Exception as e:
    log("EXCEPTION: " + traceback.format_exc())

log("=== done ===")

File: gen_tools/gen_image.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const KEY   = process.env.GAPI_KEY   || 'AIzaSyC7umcmAdwscOCeBw8xeytqyABW_FJ9YiQ';
const MODEL = process.env.GAPI_MODEL || 'imagen-4.0-generate-001';
const OUT   = process.env.GAPI_OUT   || path.join(__dirname, 'horse_sprite.png');

const PROMPT =
  'horizontal sprite sheet, exactly 6 frames in one single row from left to right, ' +
  'no vertical stacking, all frames on the same horizontal line, ' +
  'white thoroughbred racehorse with jockey riding on top, ' +
  'jockey wearing vibrant red and gold silk racing uniform and helmet, ' +
  'detailed fantasy illustration style, flowing mane and tail, ' +
  'NO wings NO horn, realistic horse anatomy, ' +
  'gallop animation cycle 6 distinct poses: contact, loading, takeoff, flight, extension, landing, ' +
  'side view facing right, pure solid bright lime green background, chroma key green #00FF00, ' +
  'flat solid color background no gradients no shadows on background, ' +
  'sharp clean outlines, game asset style, 2D illustration, ' +
  'each frame same width, evenly spaced, clear separation between frames';

const body = JSON.stringify({
  instances: [{ prompt: PROMPT }],
  parameters: { sampleCount: 1 }
});

const options = {
  hostname: 'generativelanguage.googleapis.com',
  path: `/v1beta/models/${MODEL}:predict?key=${KEY}`,
  method: 'POST',
  headers: { 'Content-Type': 'application/json', 'Content-Length': Buffer.byteLength(body) }
};

console.log('Generating horse sprite...');
const req = https.request(options, res => {
  let data = '';
  res.on('data', c => data += c);
  res.on('end', () => {
    try {
      const json = JSON.parse(data);
      if (json.predictions) {
        const img = Buffer.from(json.predictions[0].bytesBase64Encoded, 'base64');
        fs.writeFileSync(OUT, img);
        console.log(`OK - saved ${OUT} (${img.length} bytes)`);
      } else {
        console.log('API ERROR:', JSON.stringify(json, null, 2));
      }
    } catch(e) {
      console.log('Parse error:', e.message);
      console.log('Raw:', data.slice(0, 500));
    }
  });
});
req.on('error', e => console.log('Request error:', e.message));
req.write(body);
req.end();
